tracing-subscriber/src/registry/sharded.rs

```diff
@@ -0,0 +1,244 @@
+use sharded_slab::{Guard, Slab};
+
+use crate::{
+    fmt::span::SpanStack,
+    registry::{
+        extensions::{Extensions, ExtensionsInner, ExtensionsMut},
+        LookupMetadata, LookupSpan, SpanData,
+    },
+    sync::RwLock,
+};
+use std::{
+    cell::RefCell,
+    sync::atomic::{fence, AtomicUsize, Ordering},
+};
+use tracing_core::{
+    dispatcher,
+    field::FieldSet,
+    span::{self, Current, Id},
+    Event, Interest, Metadata, Subscriber,
+};
+
+/// `Registry` is a shared, reusable store for spans.
+#[derive(Debug)]
+pub struct Registry {
+    spans: Slab<Data>,
+}
+
+/// The [Registry]'s view of a span. This definition
```

Comment:
Let's use URL links rather than intra-doc links, my experience is that those don't always work correctly. Can we put 
```markdown
[Registry]: struct.Registry.html
```
at the bottom of the doc comment?

Reply:
Also, can we put backticks on "Registry"?
```suggestion
/// The [`Registry`]'s view of a span. This definition
```

Reply:
Finally, i think idiomatic Rustdoc has a single introductory sentence on its own line, followed by a blank line before additional details. So, can this be changed to:
```rust
/// The [`Registry`]'s view of a span.
///
/// This definition...